backend/app/tests/services/test_vector_db_context_service.py

```diff
@@ -128,7 +129,65 @@ def test_query_vector_db_complex(
 
     # Assert
     mock_analyze.assert_not_called()
-    mock_retriever_builder.assert_called_once_with(populate_db=False)
+    mock_retriever_builder.assert_called_once()
     fake_retriever.invoke.assert_called_once_with(expected_query)
     assert text == doc_content_1 + '\n\n' + doc_content_2
     assert metadata == [doc_metadata_1, doc_metadata_2]
+
+
+@patch('app.services.vector_db_context_service.query_vector_db')
+def test_query_vector_db_and_prompt_with_results(mock_query_vector_db: MagicMock) -> None:
+    # Set up args and mocks
+    generated_object = 'objectives'
+    session_context = ['Example context']
+    transcript = "Here's a user statement."
+
+    doc1 = 'Follow the company policy.'
+    doc2 = 'Ensure respectful communication.'
+    combined_docs = f'{doc1}\n\n{doc2}'
+    mock_query_vector_db.return_value = (combined_docs, [{'source': 'doc1'}, {'source': 'doc2'}])
+
+    result = query_vector_db_and_prompt(
+        generated_object=generated_object,
+        session_context=session_context,
+        user_transcript=transcript,
+    )
+
+    # Assert
+    expected = (
+        f'\nThe objectives you generate should comply with '
+        f'the following HR Guideline excerpts:\n'
+        f'{combined_docs}\n'
+    )
+    mock_query_vector_db.assert_called_once()
+    assert result == expected
+
+
+@patch('app.services.vector_db_context_service.query_vector_db')
+def test_query_vector_db_and_prompt_with_none(mock_query_vector_db: MagicMock) -> None:
+    # Set up args and mocks
+    generated_object = 'objectives'
+    mock_query_vector_db.return_value = (None, None)
+
+    result = query_vector_db_and_prompt(
+        generated_object=generated_object, session_context=None, user_transcript=None
+    )
+
+    # Assert
+    assert result == ''
+    mock_query_vector_db.assert_called_once()
+
+
+@patch('app.services.vector_db_context_service.query_vector_db')
+def test_query_vector_db_and_prompt_with_empty_docs(mock_query_vector_db: MagicMock) -> None:
+    # Set up args and mocks
+    generated_object = 'objectives'
+    mock_query_vector_db.return_value = ('', [])
+
+    result = query_vector_db_and_prompt(
+        generated_object=generated_object, session_context=None, user_transcript=None
+    )
+
+    # Assert
+    assert result == ''
+    mock_query_vector_db.assert_called_once()
```

Comment:
Sorry I only just noticed it and not in the previous PR: In other files, such as test_scenario_preparation_service_db, we use `assertEqual(x,y)`and not assert x==y. Would be great if you could change this too in all tests. If it is too much work, we can just do a separate cleanup PR.